react/src/components/BAIPropertyFilter.tsx

```diff
@@ -0,0 +1,193 @@
+
+interface FilterInput {
+  property: string;
+  operator: string;
+  value: string;
+  propertyLabel: string;
+}
+
+const DEFAULT_OPERATOR = 'ilike';
+
+function trimFilterValue(filterValue: string): string {
+  return filterValue.replace(/^%|%$/g, '');
+}
+
+export function parseFilterValue(filter: string) {
+  const [property, ...rest] = filter.split(/\s+(?=(?:(?:[^"]*"){2})*[^"]*$)/);
+  const [operator, ...valueParts] = rest
+    .join(' ')
+    .split(/\s+(?=(?:(?:[^"]*"){2})*[^"]*$)/);
+  const value = valueParts.join(' ').replace(/^"|"$/g, '');
+  return { property, operator, value };
+}
+
+const BAIPropertyFilter: React.FC<BAIPropertyFilterProps> = ({
+  filterProperties,
+  value: propValue,
+  onChange: propOnChange,
+  defaultValue,
+  ...otherProps
+}) => {
+  const [search, setSearch] = useControllableValue({});
+
+  const [value, setValue] = useControllableValue<string | undefined>({
+    value: propValue,
+    defaultValue: defaultValue,
+    onChange: propOnChange,
+  });
+  const filtersFromValue = useMemo(() => {
+    if (value === undefined) return [];
+    const filters = value.split('&').map((filter) => filter.trim());
+    return filters.map((filter) => {
+      const { property, operator, value } = parseFilterValue(filter);
+      return {
+        property,
+        operator,
+        value,
+        propertyLabel:
+          _.find(filterProperties, (f) => f.key === property)?.propertyLabel ||
+          property,
+      };
+    });
+  }, [value, filterProperties]);
+
+  const { t } = useTranslation();
+  const options = _.map(filterProperties, (filterProperty) => {
+    return {
+      label: filterProperty.propertyLabel,
+      value: filterProperty.key,
+      filter: filterProperty,
+    };
+  });
+  const [selectedProperty, setSelectedProperty] = useState(options[0].filter);
+  const { list, remove, push, resetList, getKey } =
+    useDynamicList<FilterInput>(filtersFromValue);
+
+  const { token } = theme.useToken();
+
+  useEffect(() => {
+    if (list.length === 0) {
+      setValue(undefined);
+    } else {
+      setValue(
+        _.map(list, (item) => {
+          return `${item.property} ${item.operator} "${item.value}"`;
+        }).join(' & '),
+      );
+    }
+  }, [list, setValue]);
+
+  return (
+    <Flex direction="column" gap={'xs'} style={{ flex: 1 }} align="start">
+      <Flex>
+        <Input.Search
+          addonBefore={
+            <Select
+              popupMatchSelectWidth={false}
+              options={options}
+              value={selectedProperty.key}
+              onChange={(value, options) => {
+                setSelectedProperty(_.castArray(options)[0].filter);
+              }}
+            />
+          }
+          style={{ minWidth: 250 }}
+          onSearch={(value) => {
+            if (_.isEmpty(value)) return;
+            setSearch('');
+            const operator =
+              selectedProperty.defaultOperator || DEFAULT_OPERATOR;
+            const filterValue =
```

Comment:
For now we only support "like" filter, right?

Reply:
Thank you! @lizable 

Yes, currently it only supports one operation for each property. However, you can set the default operator for each property. The default is `ilkie`.